docs/DeveloperGuide.adoc

```diff
@@ -12,255 +12,359 @@ ifdef::env-github[]
+image::SuggestionClassDiagram.png[]
+
+*API* :
+link:{repoURL}blob/master/src/main/java/com/notably/logic/suggestion/SuggestionEngine.java[`SuggestionEngine.java`]
+
+`SuggestionEngine` gives users the meaning of the command they input and a list of notes suggestions that they want to
+open/ delete/ search/ edit.
+
+1. `Logic` uses the `SuggestionEngineImpl` class, which implements `SuggestionEngine` interface, to parse the user command.
+2. According to the command the user inputs, `SuggestionEngineImpl` will create a `XYZSuggestionCommandParser` object
+which implements `SuggestionCommandParser` interface.
+3. This results in the creation of `XYZSuggestionCommand` object which implements `SuggestionCommand` interface.
+`XYZSuggestionCommand` is then executed by the `LogicManager`.
+4. The command execution can affect the `Model` (e.g. `open` command will set the response text in the model as
+"Open a note" and store a list of `SuggestionItem` in the model).
+5. The UI will then be able to retrieve the response text and list of `SuggestionItem` from the model to be displayed
+to the user.
+
+Given below is the Sequence Diagram for interactions within the `Logic` and `Suggestion` component for the `execute("open /")` API call.
+
+.Interactions Inside the Logic and Suggestion Component for the `open /` Command
+image::SuggestionSequenceDiagram.png[]
+
+NOTE: The lifeline for `OpenSuggestionCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+
+[[Design-CorrectionEngine]]
+==== CorrectionEngine component
+
+.Class Diagram of the CorrectionEngine Component
+image::CorrectionEngineClassDiagram.png[]
+
+The `CorrectionEngine` component revolves around two _API_ s, namely:
+
+
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+
+* The link:{repoURL}/src/main/java/com/notably/logic/correction/CorrectionEngine.java[`CorrectionEngine`] interface, implemented by `StringCorrectionEngine` and `AbsolutePathCorrectionEngine`. Concrete implementations of `CorrectionEngine` are employed to correct an uncorrected user input.
+* The link:{repoURL}/src/main/java/com/notably/logic/correction/distance/EditDistanceCalculator.java[`EditDistanceCalculator`] interface, implemented by `LevenshteinDistanceCalculator`. Concrete implementations of `EditDistanceCalculator` are employed to calculate the https://en.wikipedia.org/wiki/Edit_distance[edit distance] between two strings.
+
+
+Given below is the Sequence Diagram for interactions within the `StringCorrectionEngine` (one concrete implementation of `CorrectionEngine`) component for the `correct("uncorrected")` API call.
+
+.Interactions inside the StringCorrectionEngine component for the `correct("uncorrected")` call
+image::StringCorrectionEngineSequenceDiagram.png[]
+
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
```

Comment:
Do you intend for the relationship between BlockTree and BlockTreeItem to be composition?